src/mame/drivers/tetrisp2.cpp

```diff
@@ -1860,53 +1839,174 @@ TIMER_DEVICE_CALLBACK_MEMBER(stepstag_state::field_cb)
 	// irq 4 is definitely a 30 Hz-ish here as well,
 	// except we have a multi-screen arrangement setup and no way to pinpoint source
 	m_subcpu->set_input_line(4, HOLD_LINE);
+	stepstag_state::simulate_pc();	// any better place to put it into schedule?
```

Comment:
From what I can gather:
- the main2pc call to `simulate_pc` can't be instant: should call an explicit emu_timer that simulates the command wait time;
- the command itself must have an handshake bit somewhere just right after it tries to send a command to the bus. The busy bit must be set/acknowledged accordingly.
- if it really expect periodic callbacks then should be a custom timer periodic declared in machine_config, with a 30 Hz resolution;

Reply:
removed at this time